geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/IngestCommand.scala

```diff
@@ -27,19 +27,32 @@ class IngestCommand(parent: JCommander) extends Command(parent) with LazyLogging
   override val params = new IngestParameters()
 
   override def execute(): Unit = {
-    val fmt = Formats.fromString(Option(params.format).getOrElse(getFileExtension(params.files(0))))
+    val extensions = params.files.map(getFileExtension)
+    require(extensions.tail.forall(_ == extensions.head), "Input files must all be of the same file type")
+    ensureSameFs(Seq("hdfs", "s3n", "s3a"))
+
+    val fmt = Formats.fromString(Option(params.format).getOrElse(extensions.head))
     if (fmt == SHP) {
       val ds = new DataStoreHelper(params).getDataStore()
-      GeneralShapefileIngest.shpToDataStore(params.files(0), ds, params.featureName)
+      params.files.foreach(GeneralShapefileIngest.shpToDataStore(_, ds, params.featureName))
     } else {
-      ensureSameFs(Seq("hdfs", "s3n", "s3a"))
-
       val dsParams = new DataStoreHelper(params).paramMap
-      require(DataStoreFinder.getDataStore(dsParams) != null, "Could not load a data store with the provided parameters")
-      val sft = CLArgResolver.getSft(params.spec, params.featureName)
-      val converterConfig = CLArgResolver.getConfig(params.config)
+      val tryDs = DataStoreFinder.getDataStore(dsParams)
+      require(tryDs != null, "Could not load a data store with the provided parameters")
+      tryDs.dispose()
 
-      new ConverterIngest(dsParams, sft, converterConfig, params.files, params.threads).run()
+      if (params.spec == null && params.config == null && Seq(TSV, CSV, AVRO).contains(fmt)) {
```

Comment:
picky - Might be useful to have a comment here explaining AutoIngest when spec/config are null